shell/browser/api/electron_api_utility_process.cc

```diff
@@ -283,34 +70,34 @@ UtilityProcessWrapper::UtilityProcessWrapper(
   base::FileHandleMappingVector fds_to_remap;
 #endif
   for (const auto& [io_handle, io_type] : stdio) {
-    if (io_type == IOType::PIPE) {
+    if (io_type == IOType::IO_PIPE) {
 #if BUILDFLAG(IS_WIN)
       HANDLE read = nullptr;
       HANDLE write = nullptr;
-      if (!::CreatePipe(&read, &write, nullptr, 0)) {
+      SECURITY_ATTRIBUTES sa_attr;
+      // Set the bInheritHandle flag so pipe handles are inherited.
+      sa_attr.nLength = sizeof(SECURITY_ATTRIBUTES);
+      sa_attr.bInheritHandle = TRUE;
```

Comment:
!, doesn't this mean that this pipe will be inherited by child processes that are spawned? it seems surprising that this would be the right choice here

Reply:
Nope, that is why we make this change https://github.com/electron/electron/blob/014d6e8b897fedba1222a36de0227e40b2d606eb/shell/common/node_bindings.cc#L396-L399 so that [uv_disable_stdio_inheritance](https://github.com/libuv/libuv/blob/47e0c5c575e92a25e0da10fc25b2732942c929f3/src/win/process-stdio.c#L74-L95) gets called in the child process.

Reply:
Why do we enable handle inheritance only to disable it again immediately afterwards?

Reply:
Update the comment in https://github.com/electron/electron/pull/34980/commits/f6b160abb490742736220374ef703bc6015a102b. The inheritance attribute is needed before spawning the process so that the write handle will be duplicated into the child process and can be used there. `uv_disable_stdio_inheritance` is done so that child processes that will be spawned from the utility process by a user does not inherit the stdio handles. Also, while writing the comment noticed that `base::Launch` already sets the required inheritance bits on the required handles so the call I wrote was redundant. Updated the comment to reflect it as well.